subsys/bluetooth/host/l2cap.c

```diff
@@ -2031,7 +2031,11 @@ static int l2cap_chan_le_send_sdu(struct bt_l2cap_le_chan *ch,
 {
 	int ret;
 	size_t sent = 0;
-	size_t rem_len = buf->len;
+	size_t rem_len;
+
+	__ASSERT_NO_MSG(buf);
+
+	rem_len = buf->len;
 
 	while (buf && sent != rem_len) {
```

Comment:
With the assert I guess the `while(buf` check can be removed, as `buf` is never modified by the loop.

Reply:
Indeed. After fragment support was removed, this whole function appears useless. I'll remove it.